persim/visuals.py

```diff
@@ -280,3 +280,178 @@ def wasserstein_matching(I1, I2, matchidx, palette=None, labels=["dgm1", "dgm2"]
+
+        if len(diagrams) == 2:
+            self.plot_barcode = self._plot_H0_H1
+
+        else:
+            self.plot_barcode = self._plot_Hn
+
+        self.diagrams = diagrams
+
+    def _plot_H0_H1(self, **kwargs):
+        '''
+        parameters:
+        ===========
+            figsize: tuple
+                figure size, default=(6,6)
+
+            show: boolean
+                show the figure via plt.show()
+
+            export_png: boolean
+                write image to png data, returned as io.BytesIO() instance, default=False
+                
+            **kwargs: artist paramters for the barcodes, defaults:
+                c='grey'
+                linestyle='-'
+                linewidth=0.5
+                dpi=100 (for png export)
+
+        returns:
+        ===========
+            list of png exports or []
+        '''
+        import io
+
+        fsize = kwargs.get('figsize', (6, 6))
+        show = kwargs.get('show', True)
+        export = kwargs.get('export_png', False)
+        dpi = kwargs.get('dpi', 100)
+
+        out = []
+
+        fig, ax = plt.subplots(2, 1, figsize=fsize)
+
+        for dim, diagram in enumerate(self.diagrams):
+            self._plot_many_bars(dim, diagram, dim, ax, **kwargs)
+
+        if export:
+            fp = io.BytesIO()
+            plt.savefig(fp, dpi=dpi)
+            fp.seek(0)
+
+            out += [fp]
+
+        if show:
+            plt.show()
+
+        return out
+
+    def _plot_Hn(self, **kwargs):
+        '''
+        parameters:
+        ===========
+            figsize: tuple
+                figure size, default=(6,6)
+
+            show: boolean
+                show the figure via plt.show()
+
+            export_png: boolean
+                write image to png data, returned as io.BytesIO() instance, default=False
+                
+            **kwargs: artist paramters for the barcodes, defaults:
+                c='grey'
+                linestyle='-'
+                linewidth=0.5
+                dpi=100 (for png export)
+
+        returns:
+        ===========
+            list of png exports or []
+        '''
+        import io
+        
+        fsize = kwargs.get('figsize', (6, 4))
+        show = kwargs.get('show', True)
+        export = kwargs.get('export_png', False)
+        dpi = kwargs.get('dpi', 100)
+
+        out = []
+
+        for dim, diagram in enumerate(self.diagrams):
+            fig, ax = plt.subplots(1, 1, figsize=fsize)
+
+            self._plot_many_bars(dim, diagram, 0, [ax], **kwargs)
+
+            if export:
+                fp = io.BytesIO()
+                plt.savefig(fp, dpi=dpi)
+                fp.seek(0)
+
+                out += [fp]
+
+            if show:
+                plt.show()
+
+        return out
+
+    def _plot_many_bars(self, dim, diagram, idx, ax, **kwargs):
+        number_of_bars = len(diagram)
+        print("Number of bars in dimension %d: %d" % (dim, number_of_bars))
```

Comment:
can you thread a `verbose` boolean argument through these functions? Preferably as a parameter to the class.

Then, wrap all of the print statements in a 
```
if self._verbose: 
    print(...)
```